daffodil-core/src/main/scala/org/apache/daffodil/dsom/Nesting.scala

```diff
@@ -54,24 +61,61 @@ trait NestingMixin {
  * schema components.
  */
 trait NestingLexicalMixin
-  extends NestingMixin {
+  extends NestingMixin { self: SchemaComponent =>
 
+  @deprecated("2019-06-03", "Use enclosingComponentDefs method, and deal with sharing.")
   override protected def enclosingComponentDef = optLexicalParent
 
+  final override protected lazy val enclosingComponentDefs: Seq[(SchemaComponent, Int)] =
```

Comment:
Thoughts on creating a new type instead of using a tuple here? Might make things a bit more readable, something like
```scala
case class EnclosingComponentDef(lexicalParent: SchemaComponent, lexicalPosition: Int)
```
Not sure if there's a benefit or not, but might help readablility, especially once everything gets rewritten to using enclosingComponentDefs.

Reply:
I will incorporate this idea.